app/scripts/modules/core/cancelModal/cancelModal.controller.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+let angular = require('angular');
+
+require('./cancel.less');
+
+module.exports = angular
+  .module('spinnaker.core.cancelModal.controller', [
+    require('angular-ui-bootstrap')
+  ])
+  .controller('CancelModalCtrl', function($scope, $uibModalInstance, params) {
+    $scope.params = params;
+
+    $scope.state = {
+      submitting: false
+    };
+
+    this.formDisabled = () => $scope.state.submitting;
+
+    function showError(exception) {
+      $scope.state.error = true;
+      $scope.state.submitting = false;
+      $scope.errorMessage = exception;
+    }
+
+    this.confirm = () => {
+      if (!this.formDisabled()) {
```

Comment:
You will also need to check the submitting state here because it's possible for the UI to not be updated before the user [accidentally] clicks the button a second time - rendering your `ng-disabled` worthless.